### PR TITLE
Rename Contributions to Containers

### PR DESCRIPTION
**Pull request by Coder206**
This PR addresses part of #59, renaming Contributions to Containers and also redirecting from contributions to containers. 

Upon scanning the repository with Cloud 9, I found that `/dockerfiles/ubuntu-dev/client-workspace-janitor.js` has a [reference](https://github.com/JanitorTechnology/dockerfiles/blob/master/ubuntu-dev/workspace-janitor.js#L27) to contributions but I was unsuccessful in adding it to the PR. Would it be better to commit this to the [dockerfiles repository](https://github.com/JanitorTechnology/dockerfiles/)?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -237,8 +237,21 @@ boot.executeInParallel([
       return;
     }
 
-    routes.contributionsPage(response, user);
+    routes.containersPage(response, user);
   });
+  
+  // User container list.
```
Comment (nt1m) [inline app.js]: nit: containers

File: app.js [hunk 2]
```diff
@@ -228,7 +228,7 @@ boot.executeInParallel([
     });
   });
 
-  // User contributions list.
+  // User contributions list. (legacy - redirect to containers page)
   app.route(/^\/contributions\/?$/, (data, match, end, query) => {
```
Comment (nt1m) [inline app.js]: I think it would be better the make the regex below (containing containers) to include contributions rather than duplicate the block.

Comment (Coder206) [reply app.js]: That makes sense would the resulting syntax be `/^\/containers|contributions\/?$/`?

**Comment by Coder206** [comment]
@nt1m Thanks for the review and suggestions, in little time delay!

**Comment by jankeromnes** [comment]
Thanks a lot @Coder206 for this pull request! And also @nt1m for the super quick review :D

> Upon scanning the repository with Cloud 9, I found that /dockerfiles/ubuntu-dev/client-workspace-janitor.js has a reference to contributions but I was unsuccessful in adding it to the PR. Would it be better to commit this to the dockerfiles repository?

Whoa, good catch! Indeed you found https://github.com/JanitorTechnology/dockerfiles/blob/448bef2ee08991e1937f3aad8875075fcfb14638/ubuntu-dev/workspace-janitor.js#L27 which needs to be modified in the dockerfiles repository directly.

However, maybe let's for the rename to be successfully merged and deployed on Janitor before updating it in dockerfiles (plus the Dashboard URL is a minor feature in Cloud9, and "contributions" will redirect to "containers" anyway, so we can take our time there).

File: app.js (hunk 2 above)
Comment (jankeromnes) [reply app.js]: Interesting suggestion! It does simplify the code somewhat, however the behavior isn't the same:

- The original approach made a permanent HTTP redirection from /contributions/ to /containers/
- The new approach makes the server reply with the same content for both /contributions/ and /containers/

What I find strange in the second approach is that we don't indicate that the page URL was changed. I agree that it's a very minor difference from the user's perspective, however if you still use the old URL (e.g. via bookmarks or browsing history), you'll never see the new URL in your URL bar.

I think I'd prefer a permanent HTTP redirection (just like we do for API requests without a final `/`; the account page; favicon.ico and apple-touch-icons; and various others).

Comment (Coder206) [reply app.js]: Unless we can notify the user about the URL change I think the permanent HTTP redirection might be best for now.

**Comment by jankeromnes** [comment]
Successfully deployed to https://janitor.technology ! Looks much cleaner now. 😄

**Pull request merged**